qdrant-landing/content/articles/vector-search-resource-optimization.md

```diff
@@ -0,0 +1,598 @@
+    search_params=models.SearchParams(
+        quantization=models.QuantizationSearchParams(
+            rescore=True,   # Enables rescoring with original vectors
+            oversampling=2  # Retrieves extra candidates for rescoring
+        )
+    ),
+    limit=4  # Desired number of final results
+)
+```
+---
+Learn more about [**Rescoring**](/articles/what-is-vector-quantization/#3-rescoring-with-original-vectors).
+
+---
+
+#### Reranking
+
+Reranking adjusts the order of search results based on additional criteria, ensuring the most relevant results are prioritized. 
+
+This method is about taking the results from different search methods and reordering them based on some additional processing using the content of the documents, not just the scores. This processing may rely on an additional neural model, such as a cross-encoder which would be inefficient enough to be used on the whole dataset. 
+
+<img src="/articles_data/vector-search-resource-optimization/reranking.png" alt="reranking" style="width: 75%;">
+
+These methods are practically applicable only when used on a smaller subset of candidates returned by the faster search methods. Late interaction models, such as ColBERT, are way more efficient in this case, as they can be used to rerank the candidates without the need to access all the documents in the collection.
+
+**Example:**
+
+```python
+client.query_points(
+       "collection-name",
+       prefetch=prefetch, # Previous results
+       query=late_vectors, # Colbert converted query
+       using="colbertv2.0",
+       with_payload=True,
+       limit=10,
+)
+___
+Learn more about [**Reranking**](/documentation/search-precision/reranking-hybrid-search/#rerank).
+
+---
+
+## Storage: Disk vs RAM
+![disk](/articles_data/vector-search-resource-optimization/disk.png)
+
+| Storage  | Description                                                                                                      |
+|--------------|------------------------------------------------------------------------------------------------------------------|
+| **RAM**      | Crucial for fast access to frequently used data, such as indexed vectors. The amount of RAM required can be estimated based on your dataset size and dimensionality. For example, storing **1 million vectors with 1024 dimensions** would require approximately **5.72 GB of RAM**. |
+| **Disk**     | Suitable for less frequently accessed data, such as payloads and non-critical information. Disk-backed storage reduces memory demands but can introduce slight latency. |
+
+#### Which Disk Type?
+
+1. **SSD**: Recommended for optimal performance, particularly for workloads involving random reads and writes. SSDs can significantly enhance query response times when the data is stored on disk.
+2. **HDD**: While more cost-effective, HDDs are slower and can negatively impact performance, especially for large datasets or applications requiring high-speed access.
```

Comment:
Almost nobody uses HDD now. The problem of slow disks is not because it is HDD, but because it is a network-attached disk in the cloud, with huge latency

Reply:
Real HDD will be totally unusable with any workload